Makefile

```diff
@@ -19,33 +26,32 @@ ROM      := zeldaalttp.gba
 ELF      := $(ROM:.gba=.elf)
 MAP      := $(ROM:.gba=.map)
 LDSCRIPT := ldscript.txt
-SOURCES  := \
-	asm/crt0.s \
-	asm/rom1.s \
-	src/main.c \
-	src/main_2.c \
-	src/math.c \
-	src/rom2.c \
-	asm/rom2.s \
-	asm/rom2a.s \
-	asm/rom3.s \
-	src/rom3a.c \
-	asm/rom3a.s \
-	asm/rom4.s \
-	src/interface.c \
-	asm/interface.s \
-	asm/rom5.s \
-	asm/syscall.s \
-	asm/rom6.s \
-	src/newlib-libc/string/memcpy.c \
-	asm/rom8.s \
-	asm/4swords_text.s
-OFILES   := $(addsuffix .o, $(basename $(SOURCES)))
-
-# Special configurations
-src/newlib-libc/string/memcpy.o: CC1 := $(CC1_OLD)
-src/newlib-libc/string/memcpy.o: CPPFLAGS += -Isrc/newlib-libc/include
-src/newlib-libc/string/memcpy.o: CC1FLAGS := -O2
+
+SUBDIRS := \
+        src \
+        asm \
+        data
+
+C_SOURCES   := $(foreach dir, $(SUBDIRS), $(wildcard $(dir)/*.c))
+ASM_SOURCES := $(foreach dir, $(SUBDIRS), $(wildcard $(dir)/*.s))
+
+C_OBJECTS   := $(C_SOURCES:%.c=%.o)
+ASM_OBJECTS := $(ASM_SOURCES:%.s=%.o)
+
+SOURCES  := $(C_SOURCES) $(ASM_SOURCES)
+OFILES   := $(C_OBJECTS) $(ASM_OBJECTS)
+
+ifeq ($(OS),Windows_NT)
```

Comment:
I'd rather just link the .a files directly on every platform rather than making a special case for Windows.